generics/services/sunbird.js

```diff
@@ -0,0 +1,590 @@
+            }
+        }
+    })
+
+}
+
+/**
+  * Upload content data
+  * @function
+  * @name uploadContent
+  * @param file - Upload file data.
+  * @param contentId - content id.
+  * @param token - logged in token.
+  * @param contentType - content type.
+  * @param mimeType - mime type.
+  * @returns {Promise}
+*/
+
+const uploadContent = async function (file, contentId, token, contentType, mimeType) {
+
+    const contentUrl =
+        process.env.SUNBIRD_BASE_URL + CONSTANTS.endpoints.SUNBIRD_UPLOAD_CONTENT + `/${contentId}`;
+
+    return new Promise(async (resolve, reject) => {
+
+        const options = {
+            "headers": {
+                "content-type": contentType,
+                "authorization": process.env.AUTHORIZATION,
+                "x-authenticated-user-token": token,
+                "x-channel-id": process.env.SUNBIRD_ORGANISATION_ID
+            },
+            formData: {
+                "fileName": {
+                    "value": fs.createReadStream(file),
+                    "options": {
+                        contentType: mimeType
+                    }
+                }
+            }
+        };
+
+        request.post(contentUrl, options, callback);
+
+        function callback(err, data) {
+            if (err) {
+                return reject({
+                    message: CONSTANTS.apiResponses.SUNBIRD_SERVICE_DOWN
+                });
+            } else {
+                const contentData = data.body;
+                return resolve(JSON.parse(contentData));
+            }
+        }
+    })
+
+}
+
+/**
+  * Call to sunbird api's. 
+  * @function
+  * @name callToSunbird
+  * @param requestBody - Logged in user Id.
+  * @param token - Logged in user token.
+  * @param url - url of the api call.
+  * @param requestType - http request method
+  * @returns {JSON} - consist of sunbird service response
+*/
+
+function callToSunbird(requestType, url, token, requestBody = "") {
+
+    return new Promise(async (resolve, reject) => {
+        let options = {
+            "headers": {
+                "content-type": "application/json",
+                "authorization": process.env.AUTHORIZATION,
+                "x-authenticated-user-token": token
+            }
+        };
+
+        if (requestType != "GET") {
+            options['json'] = { request: requestBody };
+        }
+
+        url = process.env.SUNBIRD_BASE_URL + url;
+        if (requestType == "PATCH") {
+            request.patch(url, options, callback);
+        } else if (requestType == "GET") {
+            request.get(url, options, callback);
+        } else {
+            request.post(url, options, callback);
+        }
+
+        function callback(err, data) {
+            if (err) {
+                return reject({
+                    message: CONSTANTS.apiResponses.SUNBIRD_SERVICE_DOWN
+                });
+            } else {
+
+
+                if (data.statusCode == HTTP_STATUS_CODE.ok.status) {
```

Comment:
@rakeshSgr - Plan for handling the scenario where if sunbird doesn't respond within 3 sec, give an error if it does. Make the timeout value configurable.